docs/by-example/sqlalchemy/advanced-querying.rst

```diff
@@ -0,0 +1,267 @@
+updated periodically (default: each second). In order to synchronize that,
+refresh the table:
+
+    >>> _ = connection.execute(sa.text("REFRESH TABLE characters"))
+
+
+Aggregates: Counting and grouping
+=================================
+
+SQLAlchemy supports different approaches to issue a query with a count
+aggregate function. Take a look at the `count result rows`_ documentation
+for a full overview.
+
+CrateDB currently does not support all variants as it can not handle the
+sub-queries yet.
+
+This means that queries using ``count()`` have to be written in one of the
+following ways:
+
+    >>> session.query(sa.func.count(Character.id)).scalar()
+    2
+
+    >>> session.query(sa.func.count('*')).select_from(Character).scalar()
+    2
+
+Using the ``group_by`` clause is similar:
+
+    >>> session.query(sa.func.count(Character.id), Character.name) \
+    ...     .group_by(Character.name) \
+    ...     .order_by(sa.desc(sa.func.count(Character.id))) \
+    ...     .order_by(Character.name).all()
+    [(1, 'Arthur Dent'), (1, 'Tricia McMillan')]
+
+
+Fulltext search with MATCH predicate
+====================================
+
+Fulltext search in CrateDB is performed using the :ref:`crate-reference:predicates_match`.
+The CrateDB SQLAlchemy dialect comes with a ``match`` function, which can be used to
+search on one or multiple fields.
+
+    >>> from crate.client.sqlalchemy.predicates import match
+
+    >>> session.query(Character.name) \
+    ...     .filter(match(Character.name_ft, 'Arthur')) \
+    ...     .all()
+    [('Arthur Dent',)]
+
+To get the relevance of a matching row, an internal system column ``_score``
+can be selected. It is a numeric value which is relative to the other rows.
+The higher the score value, the more relevant the row.
+
+In most cases, ``_score`` is not part of the SQLAlchemy table definition,
+so it must be passed as a string:
+
+    >>> session.query(Character.name, sa.literal_column('_score')) \
+    ...     .filter(match(Character.quote_ft, 'space')) \
+    ...     .all()
+    [('Tricia McMillan', ...)]
+
+To search on multiple columns, pass a dictionary with columns and ``boost``
+attached. ``boost`` is a factor that increases the relevance of a column in
+respect to the other columns:
+
+    >>> session.query(Character.name) \
+    ...           .filter(match({Character.name_ft: 1.5, Character.quote_ft: 0.1},
+    ...                         'Arthur')) \
+    ...     .order_by(sa.desc(sa.literal_column('_score'))) \
+    ...     .all()
+    [('Arthur Dent',), ('Tricia McMillan',)]
+
+The ``match_type`` argument determines how a single ``query_term`` is applied
+and how the resulting ``_score`` is computed. Thus, it influences which
+documents are considered more relevant. The default selection is ``best_fields``.
+For more information, see :ref:`crate-reference:predicates_match_types`.
+
+Results are ordered by ``_score`` by default, but can be overridden by adding
+an ``order_by()`` clause.
```

Comment:
Is this really the case? Afaik we don't have a default ordering for them (or at least we don't ensure that it's kept when merging results from multiple nodes)

Reply:
All right. Then let's just remove this sentence? Or should we clarify "the other way round", like:

```suggestion
Because merging results from multiple CrateDB nodes does not yield sorted records,
you would have to add an ``order_by()`` clause in order to sort by ``_score``.
```

Reply:
I'd just mention that if you want to have the results sorted by the score you can use the `order_by` function.

Without any elaboration regarding default ordering.

Reply:
Proposal:
```suggestion
Results are not sorted by default. If you want to sort them by ``_score``, you
can use the ``order_by()`` function.
```